viewing_party/party.py

```diff
@@ -1,23 +1,193 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    #if any parameter is empty return none
+    if title and genre and rating:
+        new_movie = {"title": title, "genre": genre, "rating": rating}
+        return new_movie
+    else:
+        return None
+
+
+
+def add_to_watched(user_data, movie):
+
+    #put movie inside value list of of movies
+    user_data["watched"].append(movie)
+
+    #return user data
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    
+    #new list to add unwatched movies to
+    new_watchlist = [] 
+
+    for i in range(len(user_data["watchlist"])):
+        if title in user_data["watchlist"][i].values(): #add to watched list
+            user_data["watched"].append(user_data["watchlist"][i]) 
+        else: #remain in watchlist
+            new_watchlist.append(user_data["watchlist"][i])
+
+    user_data["watchlist"] = new_watchlist #update watchlist in user data
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+
+    #initiate running total for rating score
+    total = 0
+    for i in range(len(user_data["watched"])):
+            total += user_data["watched"][i]["rating"]

```

Comment:
A ranged loop iterates over the indices of a list but in this case, we can iterate over the values and grab the rating directly like so:

```suggestion
    for movie in user_data["watched"]:
            total += movie["rating"]
```